docs/basics/101-130-yodaproject.rst

```diff
@@ -0,0 +1,600 @@
+midterm project.
+You start by creating an analysis dataset. Let's do it as a subdataset
+of ``DataLad-101``. Remember to specify the ``--dataset`` option within
+:command:`datalad create` to link it as a subdataset!
+After the last lecture, you naturally want your dataset to follow the YODA
+principles. As a start, you use the ``cfg_yoda`` procedure to help you
+structure the dataset [#f1]_.
+
+.. runrecord:: _examples/DL-101-130-103
+   :language: console
+   :workdir: dl-101/iris_data
+
+   # go back into DataLad-101
+   $ cd ../DataLad-101
+   $ datalad create -c yoda --dataset . midterm_project
+
+.. index:: ! datalad command; datalad subdatasets
+
+The :command:`datalad subdatasets` command can report on which subdatasets exist for
+``DataLad-101``. This helps you verify that the command succeeded and the
+dataset was indeed linked as a subdataset to ``DataLad-101``:
+
+.. runrecord:: _examples/DL-101-130-104
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ datalad subdatasets
+
+Not only the ``longnow`` subdataset, but also the newly created
+``midterm_project`` subdataset are displayed -- wonderful!
+
+As a next step you take care of linking your raw dataset
+adequately to your ``midterm_project`` dataset by installing it as a
+subdataset. Make sure to install it as a subdataset of ``midterm_project``,
+and not ``DataLad-101``!
+
+.. runrecord:: _examples/DL-101-130-105
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+
+   # we are in midterm_project, thus -d . points to the root of it.
+   $ datalad install -d . --source ../../iris_data input/
+
+After this directory is created, the directory structure of ``DataLad-101``
+looks like this:
+
+.. runrecord:: _examples/DL-101-130-106
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+
+   $ cd ../
+   $ tree -d
+
+Now that you have an ``input/`` directory with data, and a ``code/`` directory
+(created by the YODA procedure) for your scripts, it is time to work on the script
+for your analysis. Within ``midterm_project``, the ``code/`` directory is where
+you want to place your scripts. Finally you can try out the Python API of DataLad!
+
+But first, you plan your research question. You decide to do a
+classification analysis with a k-nearest neighbors algorithm [#f2]_. The iris
+dataset works well for such questions. Based on the features of the flowers
+(sepal and petal width and length) you will try to predict what type of
+flower (*Setosa*, *Versicolor*, or *Virginica*) a particular flower in the
+dataset is. You settle on two objectives for your analysis:
+
+#. Explore and plot the relationship between variables in the dataset and save
+   the resulting graphic as a first result.
+#. Perform a k-nearest neighbour classification on a subset of the dataset to
+   predict class membership (flower type) of samples in a left-out test set.
+   Your final result should be a statistical summary of this prediction.
+
```

Comment:
Maybe here, or right at the top of the chapter put a note that says something like "even if you don't know Python you can just c&p the code and follow along"

Reply:
Good idea, thx